src/openrct2/actions/CheatSetAction.cpp

```diff
@@ -512,17 +512,10 @@ void CheatSetAction::FixBrokenRides() const
 {
     for (auto& ride : GetRideManager())
     {
-        if ((ride.mechanic_status != RIDE_MECHANIC_STATUS_FIXING)
-            && (ride.lifecycle_flags & (RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN)))
+        if (ride.lifecycle_flags & (RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN))
         {
-            auto mechanic = RideGetAssignedMechanic(ride);
-            if (mechanic != nullptr)
-            {
-                mechanic->RemoveFromRide();
-            }
```

Comment:
These should have been kept nonetheless, shouldn't it? Otherwise the ride will be fixed but the mechanic would stay there. It also prevents the removal of a mechanic that was just inspecting.

Reply:
> These should have been kept nonetheless, shouldn't it? Otherwise the ride will be fixed but the mechanic would stay there. It also prevents the removal of a mechanic that was just inspecting.

What happens is the mechanic will continue fixing or inspecting then carry on with their duties. By just removing the check around if the ride is currently being fixed, this means the mechanic gets stuck and extra logic is needed.

Reply:
We would still want to cancel the mechanic if he is responding to a breakdown call, or heading for the ride to fix it.